CONTRIBUTING.md

```diff
@@ -0,0 +1,166 @@
+
+1. As default, GitHub Actions will be disabled on your fork. You will need to enable the GitHub Actions workflows by going to the Actions tab on your fork and clicking "I understand my workflows, go ahead and enable them".
+1. To run the "Main Build Orchestrator" on your fork, you will need to configure some repository secrets and variables. Follow the section below [Configure repository secrets and variables](#configure-repository-secrets-and-variables).
+1. Trigger an initial workflow run of the "Main Build Orchestrator". You can do this by going to the Actions tab, selecting "Main Build Orchestrator" from the list of workflows on the left (you may need to click "Show more workflows..."), then clicking "Run workflow" on the right.
+
+### Configure repository secrets and variables
+
+To run certain workflows on your fork, you will need to configure some repository secrets and variables so that the workflows can access them. These secrets and variables are used for things such as authenticating to the GitHub Container Registry and supplying Maven and Gradle builds with your GPG key information to sign artifacts.
+
+#### Repository secrets and variables that required configuring
+
+##### Repository variables
+
+To set a repository variable:
+1. Navigate to your repository settings.
+1. Select 'Secrets and variables', then 'Actions' from the menu.
+1. Click the 'Variables' tab.
+1. Select 'New repository variable'.
+1. Enter the variable name and value, then click 'Add variable'.
+
+**Required repository variables:**
+
+1. WRITE_GITHUB_PACKAGES_USERNAME: This requires the GitHub username you want to use to log into GitHub Container Registry.
+
+##### Repository secrets
+
+To set a repository variable:
+1. Navigate to your repository settings.
+1. Select 'Secrets and variables', then 'Actions' from the menu.
+1. Click the 'Secrets' tab.
+1. Select 'New repository secret'.
+1. Enter the secret name and value, then click 'Add secret'. Once you have done this, you will not be able to view the secret value again.
+
+###### Required repository secrets:
+
+1. GPG_KEY: This requires your Base 64 encoded GPG key payload.
+1. GPG_KEYID: This requires the ID of your GPG key in plain text.
+1. GPG_PASSPHRASE: This requires your passphrase for your GPG key in plain text.
+1. WRITE_GITHUB_PACKAGES_TOKEN: This requires your GitHub Personal Access Token with write:packages scope that you want to use to log into GitHub Container Registry.
+
+For the GPG related secrets, you can either use an existing key, or create a new key.
```

Comment:
Note: The Github web interface allows the setting of secrets, but you can't read them back, so be very careful what you set them to so you avoid some painful problem diagnosis steps later.

Reply:
done